### ISSUE TITLE
Moving Owncast from Github to Forgejo

### ISSUE THREAD

**Issue by janWilejan** [owncast/owncast#3176]
## Why we should move Owncast to ForgeFed / Forgejo

Currently you need a Github account to report bugs, request features, or submit pull requests to Owncast. Github is not a great company, and it should not be a prerequisite for contributing to Owncast. If we moved to a new platform, contributors would also need an account on that platform *unless we move to a federated platform*.

[ForgeFed](https://forgefed.org/) is a new ActivityPub extension for software forges. If Owncast were hosted on a ForgeFed instance, people could submit pull requests from any Vervis or Forgejo instance. In theory, users will be able to open and discuss issues from any Mastodon instance (it's not clear what that will look like or whether that also means Akkoma/Calkey/etc).

[Forgejo](https://forgejo.org/) is a fork of Gitea which is implementing ForgeFed. It is used in production by Codeberg. ForgeFed support in Forgejo is not stable yet, but we can start planning the migration process before federation stabilizes.

## The migration process:

Forgejo's federation is expected to become stable [by June 2024](https://forgefriends.org/blog/2023/06/21/2023-06-state-forge-federation/); Owncast should be an early adopter of ForgeFed and put its code on the fediverse shortly thereafter.

- Create a Forgejo instance at git.owncast.online
- Convert Github Actions to Forgejo Actions and ensure they run properly.
	- Dependency upgrading
	- Chromatic
	- Cypress
	- tests
	- lint
	- build scripts
	- deploy scripts
	- etc...
- Update Owncast's website and documentation to point to the new repository
- Owncast's Github page becomes a mirror

**Comment by gabek** [comment]
Couple more things just since they come to mind.

- All Owncast releases also pull from Github via the installer, so old versions would need to download from Github and new versions would need to pull from the new location.
- The [nightly](https://nightly.owncast.tv/) container image is pushed to the Github container registry and referenced in our k8s environment.
- Move the entire owncast.online website to another web host.

**Comment by rmens** [comment]
Do you really want to move this project, that has more ambition than contributors, to some unknown barely stable (maybe in 2024) platform? That would tank the visibility.

**Comment by janWilejan** [comment]
Forgejo is a stable platform; it's a fork of Gitea and it's used in production by Codeberg. Several fediverse projects are on Forgejo already, such as Calkey and Kbin.  The only thing unstable about it are the federated features which are not present in the latest release of Forgejo. 

I'm not quite sure what you mean by tanking visibility. Do you think people hear about Owncast because of the platform that hosts its code?

**Comment by rmens** [comment]
Thanks for the explanation. I might have read too fast and assumed the whole platform was going to be stable in June 2024. 

People hear about Owncast in other places, for sure. But what I was focussing on was the ease to contribute. There's al lot more people with a Github account and knowledge of how Github works. If Owncast was on a relatively unknown code hosting service, I wouldn't have become a contributor. With Github the friction is low, and low friction is what you want to have when attracting contributors.

**Comment by gabek** [comment]
For the record I have no intention of doing this any time soon. A year from now, two years from now? Maybe. It would make my life difficult to do so, but I'm also aware that some day Github/Microsoft will do something awful and we'll have to have a plan B. That's not a matter of if, but when.

**Comment by gabek** [comment]
> I'm not quite sure what you mean by tanking visibility. Do you think people hear about Owncast because of the platform that hosts its code?

Yes, people do discover this project through the Github social graph and it being on Github allows more people to contribute. We would absolutely have fewer contributors and visibility if we weren't on Github.

I know I don't contribute to projects if I have to create a new Gittea account. Maybe federated forges will fix that, and time will tell. So hopefully it works really well when it comes out. We'll have to see. There's a long road before we have to make any future calls.

**Comment by gabek** [comment]
Wanted to link these two threads: https://github.com/owncast/owncast/discussions/1957

<blockquote><img src="https://opengraph.githubassets.com/7b0270aff42a2dc06c8c2eb91081e764e505bf5a9286602223c754dfed0b7fea/owncast/owncast/discussions/1957" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/owncast/owncast/discussions/1957">Migration of the source code to Codeberg · owncast owncast · Discussion #1957</a></strong></div><div>Given the dependencies on Microsoft in 2018 after it bought GitHub, it makes me wonder if the source code might not also be on Codeberg, a Gitea-based GitHub and GitLab alternative?</div></blockquote>
